### PR TITLE
fix(ivy): avoid missing imports for types that can be represented as values

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Prior to this change, TypeScript stripped out some imports in case we reference a type that can be represented as a value (for ex. classes). This fix ensures that we use correct symbol identifier, which makes TypeScript retain the necessary import statements.

Thanks @alxhub for the help with investigation and fix.

This PR resolves FW-1099.

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/compiler-cli/test/ngtsc/ngtsc_spec.ts [hunk 1]
```diff
@@ -1843,6 +1843,111 @@ describe('ngtsc behavioral tests', () => {
...
+        decorators: [{
+            type: Inject,
+            args: ['arg-token']
+        }]
+      }]; }, null);
+    `;
+
+    env.driveMain();
+    const jsContents = trim(env.getContents('test.js'));
+    expect(jsContents).toContain(trim(imports));
```
Comment (alxhub) [inline packages/compiler-cli/test/ngtsc/ngtsc_spec.ts]: I'd inline the assertions here, and only assert those conditions that fall under the scope of the test. We don't need to test that the whole `setClassMetadata` call is character-for-character correct. In my experience that tends to make the tests too brittle.

File: packages/compiler-cli/test/ngtsc/ngtsc_spec.ts [hunk 2]
```diff
@@ -1843,6 +1843,111 @@ describe('ngtsc behavioral tests', () => {
...
+    expect(jsContents).toContain(trim(injectableMeta));
+    expect(jsContents).toContain(trim(componentMeta));
+  });
+
+  it('should use `undefined` in setClassMetadata if types can\'t be represented as values', () => {
+    env.tsconfig({});
+
+    env.write(`types.ts`, `
+      export interface MyType {
```
Comment (alxhub) [inline packages/compiler-cli/test/ngtsc/ngtsc_spec.ts]: Change this to use `export type MyType = Map<any, any>` - I'd like to test it with a type that does have a resolvable value side.

**Review by alxhub — CHANGES_REQUESTED** [review]

File: packages/compiler-cli/test/ngtsc/ngtsc_spec.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/compiler-cli/test/ngtsc/ngtsc_spec.ts]: Thanks Alex, I've updated the test to check for the things we need to verify in a context of this test.

**Comment by AndrewKushnir** [comment]
@alxhub thanks for the review! I've applied the changes based on your feedback. Could you please take another look? Thank you.

File: packages/compiler-cli/test/ngtsc/ngtsc_spec.ts [hunk 3]
```diff
@@ -1843,6 +1846,64 @@ describe('ngtsc behavioral tests', () => {
     expect(jsContents).toContain('ɵsetClassMetadata(TestPipe, ');
   });
 
+  it('should use imported types in setClassMetadata if they can be represented as values', () => {
```
Comment (alxhub) [inline packages/compiler-cli/test/ngtsc/ngtsc_spec.ts]: Please add a test to ensure we get it right if the class is implemented as:

```
import * as foo from 'foo';
...
constructor(f: foo.Foo) {}
```

**Review by alxhub — APPROVED** [review]
One more test is needed, otherwise lgtm!

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:235962879:BASE:235962915:1551296981393:e3abb76d)

**Pull request closed without merging**